### DISCUSSION TITLE
Is there a way to mix two bundle together ?

### DISCUSSION THREAD

**Discussion by facebreeze** [SpinalHDL/SpinalHDL#1604] [Ideas]
Hi, All,
     If BundleA and BundleB already well defined in a project, is there a way to get BundleC by mix/merge BundleA and BundleB ? 
```scala
case class BundleA() extends Bundle {
  val a = UInt(8 bits)
}

case class BundleB() extends Bundle {
  val b = UInt(8 bits)
}

case class BundleC() extends Bundle {
  val a = UInt(8 bits)
  val b = UInt(8 bits)
}
```

**Comment by Dolu1990** [comment]
Hi,

Hmmm not possible, you can only assemble them via composition

**Comment by Readon** [comment]
> Hi, All, If BundleA and BundleB already well defined in a project, is there a way to get BundleC by mix/merge BundleA and BundleB ?
> 
> ```scala
> case class BundleA() extends Bundle {
>   val a = UInt(8 bits)
> }
> 
> case class BundleB() extends Bundle {
>   val b = UInt(8 bits)
> }
> 
> case class BundleC() extends Bundle {
>   val a = UInt(8 bits)
>   val b = UInt(8 bits)
> }
> ```

How about this?
``` scala
case class BundleC() extends Bundle {
   val a = BundleA()
   val b = BundleB()
}
```
If you do not want to access the BundleA's signal inside BundleC with `a.a`,  you can define an implicit function for that.

**Reply by facebreeze** [reply]
Thanks. Both access signal in BundleC with a.a and define a implicit function looks like troublesome for me.
